spirv_glsl.cpp

```diff
@@ -2802,7 +2802,9 @@ void CompilerGLSL::emit_texture_op(const Instruction &i)
 	auto &imgtype = get<SPIRType>(type.self);
 
 	// Mark that this shader reads from this image
-	imgtype.image.is_read = true;
+	auto *p_var = maybe_get_backing_variable(img);
+	if (p_var)
+		unset_decoration(p_var->self, DecorationNonReadable);
```

Comment:
OpImageFetch is for sampled images (sampler2D), this maps to texelFetch in GLSL.
OpImageRead maps to imageLoad() (on image2D).

OpImageFetch is a bit special since it doesn't use samplers, but it only works on sampled images (so decorations do not matter). The typical sequence to for texelFetch() is
```
OpTypeSampledImage %some_sampled_image
%image_without_sampler = OpImage %some_sampled_image
%result = OpImageFetch %image_without_sampler
```

The main reason for this is that you have samplerBuffers in GLSL which are sampled images, but they don't have a sampler.